packages/core/src/lib/implementation/persist.ts

```diff
@@ -32,11 +32,13 @@ export async function persistReport(
   const { persist } = config;
   const outputDir = persist.outputDir;
   const filename = persist?.filename || reportName(report);
-  let { format } = persist;
-  format = format && format.length !== 0 ? format : ['stdout'];
-  let scoredReport;
+  const format =
+    persist.format && persist.format.length !== 0 ? persist.format : ['stdout'];
+
+  // Score the report once at the start
+  const scoredReport = scoreReport(report);
```

Comment:
If the formats are only json we do not need to score it. This is the reason I had is in each format case. This is more efficient and does not cause too much noise in the code i think.

Reply:
Agree, reverted that thing